packages/applications-service-api/src/controllers/advice.js

```diff
@@ -2,78 +2,83 @@ const { StatusCodes } = require('http-status-codes');
 
 const logger = require('../lib/logger');
 const { documentsHost } = require('../lib/config');
-const { getAdvice, getAdviceById } = require('../services/advice.service');
-
+const {
+	getAllAdvice: getAllAdviceService,
+	getAdviceById: getAdviceByIdService
+} = require('../services/advice.service');
 const ApiError = require('../error/apiError');
 
-module.exports = {
-	async getAdvice(req, res) {
-		logger.debug(`Retrieving advice ...`);
-		try {
-			const itemsPerPage = Math.min(Number(req.query.size), 100) || 25;
-			const page = Number(req.query.page) || 1;
+const getAdvice = async (req, res) => {
+	logger.debug(`Retrieving advice ...`);
+	const { caseRef } = req.query;
 
-			const advice = await getAdvice({
-				caseReference: req.query.caseRef,
-				page,
-				itemsPerPage,
-				searchTerm: req.query.searchTerm
-			});
+	try {
+		if (!caseRef) {
+			throw ApiError.badRequest('missing required parameter: caseRef');
+		}
 
-			const paginationData = {
-				totalItems: advice.count,
-				itemsPerPage: itemsPerPage,
-				totalPages: Math.ceil(Math.max(1, advice.count) / itemsPerPage),
-				currentPage: page
-			};
+		const { advice, totalItems, itemsPerPage, totalPages, currentPage } = await getAllAdviceService(
+			req.query
+		);
 
-			res.status(StatusCodes.OK).send({
-				advice: advice.rows,
-				...paginationData
-			});
-		} catch (e) {
-			if (e instanceof ApiError) {
-				logger.debug(e.message);
-				res.status(e.code).send({ code: e.code, errors: e.message.errors });
-				return;
-			}
-			logger.error(e.message);
-			res.status(StatusCodes.INTERNAL_SERVER_ERROR).send({
-				message: `Problem getting advice \n ${e}`
-			});
+		if (totalItems === 0) {
+			throw ApiError.adviceNotFound(caseRef);
 		}
-	},
 
-	async getAdviceById(req, res) {
-		const { adviceID } = req.params;
-		try {
-			logger.debug(`Retrieving advice by ID...`);
+		res.status(StatusCodes.OK).send({
+			advice: advice,
+			totalItems: totalItems,
+			itemsPerPage: itemsPerPage,
+			totalPages: totalPages,
+			currentPage: currentPage
+		});
+	} catch (e) {
```

Comment:
I think we can update this endpoint to use our newer pattern of handling errors at the router level. 

You can remove the `catch` and just `throw` errors in the controller, if you add the `asyncRoute` wrapper function in the router.

I did this for another old endpoint here: https://github.com/Planning-Inspectorate/applications-service/pull/917/files#diff-5fd35a5e8220a0d300bf5a4af2f79b81b9c7e91bb41cdb48e65fcce5725fac10R8